bip-xxxx.mediawiki

```diff
@@ -0,0 +1,358 @@
+* <code>maxadditionalfeecontribution=</code> and <code>minfeerate=</code> should be ignored in the [[#spare-change|spare change]] case.
+
+The sender must be careful to only sign the inputs that were present in the original PSBT and nothing else.
+
+Note:
+* The sender should allow the payment output to be modified by the receiver (The receiver may substitute a P2WPKH payment to P2SH payment to increase privacy)
+* The sender must allow the receiver to add outputs.
+* The sender must allow the receiver to not add any input. Useful for the receiver to change the paymout output scriptPubKey type.
+* If no input have been added, the sender's wallet implementation should accept the payjoin proposal, but not mark the transaction as an actual payjoin in the user interface.
+
+Our method of checking the fee allows the receiver and the sender to batch payments in the payjoin transaction.
+It also allows the receiver to pay the fee for batching adding his own outputs.
+
+On top of those check, it is recommended, but not required for the sender to check that:
+* If the sender is making a payjoin with a change (ie, not in the [[#spare-change|spare change]] case), make sure the receiver is paying for any batched output.
+
+===Optional parameters===
+
+When the payjoin sender posts the original PSBT to the receiver, he can optionally specify the following HTTP query string parameters:
+
+* <code>v=</code>, the version number of the payjoin protocol that the sender is using. The current version is <code>1</code>.
+
+This can be used in the future so the receiver can reject a payjoin if the sender is using a version which is not supported via an error HTTP 400, <code>version-unsupported</code>.
+If not specified, the receiver will assume the sender is <code>v=1</code>.
+
+If the receiver does not support the version of the sender, they should send an error with the list of supported versions:
+<pre>
+{
+    "errorCode": "version-unsupported",
+    "supported" : [ 2, 3, 4 ],
+    "message": "The version is not supported anymore"
+}
+</pre>
+
+* <code>additionalfeeoutputindex=</code>, the preferred output from which to increase the fee for the added inputs. (default: <code>-1</code>)
+
+If the <code>additionalfeeoutputindex</code> is out of bounds or pointing to the payment output meant for the receiver, the receiver should ignore the parameter.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>maxadditionalfeecontribution=</code>, an integer defining the maximum amount in satoshis that the sender is willing to contribute towards fees for the additional inputs. <code>maxadditionalfeecontribution</code> must be ignored if set to less than zero. (default: -1)
+
+Note that if <code>maxadditionalfeecontribution</code> is too low, the sender should create a transaction with RBF disabled, as the original transaction could replace the payjoin transaction.
```

Comment:
I don't understand this. How could the original transaction replace the payjoin transaction?
BIP-0125 rule 3 says
> The replacement transaction pays an absolute fee of at least the sum paid by the original transactions.

Why would the absolute fee of the original tx be higher than the absolute fee of the payjoin tx?
I thought it's only the feerate that might decrease?

Reply:
Check `incrementalRelayFee`. Basically nodes force RBFed transaction to pay more fees to prevent DDoS attack. Without doing this, an attacker could create an infinite number of RBFed transaction at no cost that the whole network would flood to everyone.

https://github.com/bitcoin/bitcoin/blob/371a73e94043761a6ac6ecdf77ae26cd1f289053/src/validation.cpp#L914

Reply:
I still don't see why the absolute fee of the original tx might be higher than the absolute fee of the payjoin tx.
My premise is that that does not happen.
From that AFAICT it follows that the original tx cannot replace the payjoin tx in the mempool via RBF rules.

See https://github.com/bitcoin/bitcoin/blob/371a73e94043761a6ac6ecdf77ae26cd1f289053/src/validation.cpp#L904

Reply:
The original transaction can't pay more absolute fees than the payjoin, unless the receiver did something really funky that it shouldn't be doing.

And besides, the only reason the original transaction should be broadcasted if the payjoin is a success -- is if one of the parties is being a dick. And if one of the parties wants to be a dick, there's literally nothing you can do as both parties are able to broadcast the original transaction before the payjoin even finishes.

Reply:
In that I think the remark ("sender should create a transaction with RBF disabled") from the text should be removed as it's simply confusing.